src/Domain/ElementDistribution.cpp

```diff
@@ -117,15 +118,21 @@ BlockZCurveProcDistribution<Dim>::BlockZCurveProcDistribution(
   size_t remaining_elements_in_block =
       add_number_of_elements_for_refinement(0_st, refinements_by_block[0]);
   size_t current_block = 0;
+  size_t number_of_ignored_procs_so_far = 0;
   for (size_t i = 0; i < number_of_procs; ++i) {
+    size_t current_proc_number = i + number_of_ignored_procs_so_far;
+    while (procs_to_ignore.find(current_proc_number) != procs_to_ignore.end()) {
```

Comment:
why do you need the `current_proc_number` stuff? I would've expected that all you need is:
```
if (procs_to_ignore.find(i) != procs_to_ignore.end()) {
  continue;
}
```
but if you need the extra stuff, there should be a comment explaining why. Also, wouldn't the number of procs change if you're ignoring some? At least in so far as figuring out the number of elements per proc?

Okay, I see that the next commit changes how the `number_of_procs` is passed in. I think that needs to be documented, and maybe the variable should be renamed `number_of_procs_with_elements` or something?